### DISCUSSION TITLE
Error during application compilation

### DISCUSSION THREAD

**Discussion by Emmanuel-324** [idaholab/moose#27288] [Q&A Getting Started] [answered]
Below is snippet of the error

home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addParametersToList(wasp::DataArray&, const InputParameters&, co
nst std::set<std::__cxx11::basic_string<char> >&, int, int, int, int, const std::string&)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:529:9: error: 'client_snippet_support' was not declared in this scope
  529 |     if (client_snippet_support && !default_value.empty())
      |         ^~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:531:32: error: 'm_text_format_snippet' is not a member of 'wasp::lsp'
  531 |       text_format = wasp::lsp::m_text_format_snippet;
      |                                ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:536:32: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  536 |       text_format = wasp::lsp::m_text_format_plaintext;
      |                                ^~~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:544:45: error: too many arguments to function 'bool wasp::lsp::buildCompletionObject(wasp::DataObject&,
 std::ostream&, const std::string&, int, int, int, int, const std::string&, int, const std::string&, const std::string&, bool, bool)'
  544 |     pass &= wasp::lsp::buildCompletionObject(*item,
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
  545 |                                              errors,
      |                                              ~~~~~~~
     |                                              ~~~~~~~
  546 |                                              param_name,
      |                                              ~~~~~~~~~~~
  547 |                                              replace_line_beg,
      |                                              ~~~~~~~~~~~~~~~~~
  548 |                                              replace_char_beg,
      |                                              ~~~~~~~~~~~~~~~~~
  549 |                                              replace_line_end,
      |                                              ~~~~~~~~~~~~~~~~~
  550 |                                              replace_char_end,
      |                                              ~~~~~~~~~~~~~~~~~
  551 |                                              insert_text,
      |                                              ~~~~~~~~~~~~
  552 |                                              complete_kind,
      |                                              ~~~~~~~~~~~~~~
  553 |                                              "",
      |                                              ~~~
  554 |                                              doc_string,
      |                                              ~~~~~~~~~~~
  555 |                                              false,
      |                                              ~~~~~~
  556 |                                              false,
      |                                              ~~~~~~
  557 |                                              text_format);
      |                                              ~~~~~~~~~~~~
In file included from /home/emmanuel324/projects/moose/framework/build/header_symlinks/MooseServer.h:13,
                 from /home/emmanuel324/projects/moose/framework/src/base/MooseApp.C:53,
                 from /home/emmanuel324/projects/moose/framework/build/unity_src/base_Unity.C:13:
/home/emmanuel324/projects/moose/framework/contrib/wasp/install/include/wasplsp/LSP.h:604:6: note: declared here
  604 | bool buildCompletionObject( DataObject        & object          ,
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addSubblocksToList(wasp::DataArray&, const std::string&, int, in
t, int, int, const std::string&, bool)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:618:25: error: 'client_snippet_support' was not declared in this scope
  618 |       int text_format = client_snippet_support ? wasp::lsp::m_text_format_snippet
      |                         ^~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:618:61: error: 'm_text_format_snippet' is not a member of 'wasp::lsp'
  618 |       int text_format = client_snippet_support ? wasp::lsp::m_text_format_snippet
      |                                                             ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:619:61: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  619 |                                                : wasp::lsp::m_text_format_plaintext;
      |                                                             ^~~~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:641:47: error: too many arguments to function 'bool wasp::lsp::buildCompletionObject(wasp::DataObject&,
 std::ostream&, const std::string&, int, int, int, int, const std::string&, int, const std::string&, const std::string&, bool, bool)'
  641 |       pass &= wasp::lsp::buildCompletionObject(*item,
      |               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
  642 |                                                errors,
      |                                                ~~~~~~~
  643 |                                                subblock_name,
      |                                                ~~~~~~~~~~~~~~
  644 |                                                replace_line_beg,
      |                                                ~~~~~~~~~~~~~~~~~
  645 |                                                replace_char_beg,
      |                                                ~~~~~~~~~~~~~~~~~
  646 |                                                replace_line_end,
      |                                                ~~~~~~~~~~~~~~~~~
      |                                                ~~~~~~~~~~~~~~~~~
  647 |                                                replace_char_end,
      |                                                ~~~~~~~~~~~~~~~~~
  648 |                                                insert_text,
      |                                                ~~~~~~~~~~~~
  649 |                                                complete_kind,
      |                                                ~~~~~~~~~~~~~~
  650 |                                                "",
      |                                                ~~~
  651 |                                                doc_string,
      |                                                ~~~~~~~~~~~
  652 |                                                false,
      |                                                ~~~~~~
  653 |                                                false,
      |                                                ~~~~~~
  654 |                                                text_format);
      |                                                ~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/contrib/wasp/install/include/wasplsp/LSP.h:604:6: note: declared here
  604 | bool buildCompletionObject( DataObject        & object          ,
      |      ^~~~~~~~~~~~~~~~~~~~~
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C: In member function 'bool MooseServer::addValuesToList(wasp::DataArray&, const InputParameters&, const 
std::set<std::__cxx11::basic_string<char> >&, const std::string&, const std::set<std::__cxx11::basic_string<char> >&, int, int, int, int)':
/home/emmanuel324/projects/moose/framework/src/base/MooseServer.C:804:57: error: 'm_text_format_plaintext' is not a member of 'wasp::lsp'
  804 |                                              wasp::lsp::m_text_format_plaintext);
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~
Compiling C++ (in opt mode) /home/emmanuel324/projects/moose/framework/build/unity_src/multiapps_Unity.C...
make: *** [/home/emmanuel324/projects/moose/framework/build/unity_src/base_Unity.x86_64-pc-linux-gnu.opt.lo] Error 1
make: *** Waiting for unfinished jobs....

**Comment by brandonlangley** [comment, marked as answer]
@Emmanuel-324 -

It appears that your versions of WASP and MOOSE are not in sync, and WASP just needs to be updated.

If you are using conda packages for these components, you can update to the latest version by running:

```
conda update moose-dev
```

And if that does not work, then you can try to remove and reinstall a new moose conda environment with:

```
conda env remove -n moose -all
conda create -n moose moose-dev
```

If you are not using conda and are manually installing everything, you can run this from the MOOSE root:

```
./scripts/update_and_rebuild_wasp.sh
```

I would suggest to then clean up your repo with `git clean -xfd` before retrying the MOOSE build step.

Note that `git clean -xfd` will delete any uncommitted work, so make sure you save your changes first.

**Reply by Emmanuel-324** [reply]
I am installing it through an HPC cluster. Do I follow the same process?

**Reply by brandonlangley** [reply]
@Emmanuel-324 -

You will not be using the conda package for WASP on HPC, so you will just need to run this from the root of the MOOSE repo:

```
./scripts/update_and_rebuild_wasp.sh
```

That script should pull and build the correct version of WASP needed for your version of MOOSE, then you can build MOOSE.

And if the MOOSE build fails again after this, I suggest to commit your changes, run `git clean -xfd`, then try to build again.

**Reply by Emmanuel-324** [reply]
I have git clean and rebuilt the WASP but the MOOSE build always fails

**Reply by GiudGiud** [reply]
Do you go in the top folder, moose/ before running git clean?
Do you have wasp or moose related environment variables set that would make you find the wrong moose or wasp when building?
Has the error changed since the first time you posted?
Finally what version of moose are you using?

**Reply by Emmanuel-324** [reply]
Yes i git clean in MOOSE folder. 

"module reset
module load apps site/tinkercliffs/easybuild/setup
module load apps site/tinkercliffs-rome/easybuild/setup
module load OpenMPI CMake
export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
mamba activate python-stack" This is how I load my environment.

The error has not changed

I am installing the MOOSE through an HPC cluster. Currently, I am trying to restart the whole installation process again by deleting the moose folder

**Reply by GiudGiud** [reply]
Does the Python-stack environment contain wasp by any chance? You can check with mamba list

**Reply by Emmanuel-324** [reply]
This is the list of my python stack

```pre
python-stack) [emmanuel324@tinkercliffs1 ~]$ mamba list
# packages in environment at /home/emmanuel324/mambaforge3/envs/python-stack:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
alabaster                 0.7.16                   pypi_0    pypi
asttokens                 2.4.1                    pypi_0    pypi
attrs                     23.2.0                   pypi_0    pypi
babel                     2.14.0                   pypi_0    pypi
beautifulsoup4            4.12.3                   pypi_0    pypi
bleach                    6.1.0                    pypi_0    pypi
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2023.11.17           hbcca054_0    conda-forge
certifi                   2024.2.2                 pypi_0    pypi
charset-normalizer        3.3.2                    pypi_0    pypi
contourpy                 1.2.0                    pypi_0    pypi
coverage                  7.4.4                    pypi_0    pypi
cycler                    0.12.1                   pypi_0    pypi
cython                    3.0.10                   pypi_0    pypi
decorator                 5.1.1                    pypi_0    pypi
defusedxml                0.7.1                    pypi_0    pypi
docutils                  0.20.1                   pypi_0    pypi
exceptiongroup            1.2.0                    pypi_0    pypi
executing                 2.0.1                    pypi_0    pypi
fastjsonschema            2.19.1                   pypi_0    pypi
fonttools                 4.50.0                   pypi_0    pypi
furo                      2024.1.29                pypi_0    pypi
idna                      3.6                      pypi_0    pypi
imagesize                 1.4.1                    pypi_0    pypi
importlib-resources       6.4.0                    pypi_0    pypi
iniconfig                 2.0.0                    pypi_0    pypi
ipython                   8.23.0                   pypi_0    pypi
jedi                      0.19.1                   pypi_0    pypi
jinja2                    3.1.3                    pypi_0    pypi
jsonschema                4.21.1                   pypi_0    pypi
jsonschema-specifications 2023.12.1                pypi_0    pypi
jupyter-client            8.6.1                    pypi_0    pypi
jupyter-core              5.7.2                    pypi_0    pypi
jupyterlab-pygments       0.3.0                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.2.0               h807b86a_3    conda-forge
libgomp                   13.2.0               h807b86a_3    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libsqlite                 3.44.0               h2797004_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
markupsafe                2.1.5                    pypi_0    pypi
matplotlib                3.8.3                    pypi_0    pypi
matplotlib-inline         0.1.6                    pypi_0    pypi
mistune                   3.0.2                    pypi_0    pypi
nbclient                  0.10.0                   pypi_0    pypi
nbconvert                 7.16.3                   pypi_0    pypi
nbformat                  5.10.3                   pypi_0    pypi
nbsphinx                  0.9.3                    pypi_0    pypi
nbsphinx-link             1.3.0                    pypi_0    pypi
ncurses                   6.4                  h59595ed_2    conda-forge
numpy                     1.26.4                   pypi_0    pypi
oldest-supported-numpy    2023.12.21               pypi_0    pypi
openssl                   3.1.4                hd590300_0    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
pandas                    2.2.1                    pypi_0    pypi
pandocfilters             1.5.1                    pypi_0    pypi
parso                     0.8.3                    pypi_0    pypi
pexpect                   4.9.0                    pypi_0    pypi
pillow                    10.3.0                   pypi_0    pypi
pip                       24.0                     pypi_0    pypi
platformdirs              4.2.0                    pypi_0    pypi
pluggy                    1.4.0                    pypi_0    pypi
prompt-toolkit            3.0.43                   pypi_0    pypi
ptyprocess                0.7.0                    pypi_0    pypi
pure-eval                 0.2.2                    pypi_0    pypi
pycalphad                 0.10.4.post2.dev7+g2116efe          pypi_0    pypi
pygments                  2.17.2                   pypi_0    pypi
pyparsing                 3.1.2                    pypi_0    pypi
pytest                    8.1.1                    pypi_0    pypi
pytest-cov                5.0.0                    pypi_0    pypi
python                    3.10.13         hd12c33a_0_cpython    conda-forge
python-dateutil           2.9.0.post0              pypi_0    pypi
python_abi                3.10                    4_cp310    conda-forge
pytz                      2024.1                   pypi_0    pypi
pyyaml                    6.0.1           py310h2372a71_1    conda-forge
pyzmq                     25.1.2                   pypi_0    pypi
readline                  8.2                  h8228510_1    conda-forge
referencing               0.34.0                   pypi_0    pypi
requests                  2.31.0                   pypi_0    pypi
rpds-py                   0.18.0                   pypi_0    pypi
scipy                     1.11.4                   pypi_0    pypi
setuptools                69.2.0                   pypi_0    pypi
setuptools-scm            8.0.4                    pypi_0    pypi
six                       1.16.0                   pypi_0    pypi
snowballstemmer           2.2.0                    pypi_0    pypi
soupsieve                 2.5                      pypi_0    pypi
sphinx                    7.2.6                    pypi_0    pypi
sphinx-basic-ng           1.0.0b2                  pypi_0    pypi
sphinxcontrib-applehelp   1.0.8                    pypi_0    pypi
sphinxcontrib-devhelp     1.0.6                    pypi_0    pypi
sphinxcontrib-htmlhelp    2.0.5                    pypi_0    pypi
sphinxcontrib-jsmath      1.0.1                    pypi_0    pypi
sphinxcontrib-qthelp      1.0.7                    pypi_0    pypi
sphinxcontrib-serializinghtml 1.1.10                   pypi_0    pypi
stack-data                0.6.3                    pypi_0    pypi
symengine                 0.11.0                   pypi_0    pypi
tinycss2                  1.2.1                    pypi_0    pypi
tinydb                    4.8.0                    pypi_0    pypi
tk                        8.6.13          noxft_h4845f30_101    conda-forge
tomli                     2.0.1                    pypi_0    pypi
tornado                   6.4                      pypi_0    pypi
traitlets                 5.14.2                   pypi_0    pypi
typing-extensions         4.10.0                   pypi_0    pypi
tzdata                    2024.1                   pypi_0    pypi
urllib3                   2.2.1                    pypi_0    pypi
wcwidth                   0.2.13                   pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
wheel                     0.43.0                   pypi_0    pypi
xarray                    2024.3.0                 pypi_0    pypi
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
```

**Reply by brandonlangley** [reply]
@Emmanuel-324 -

Could you copy and paste the output from these two commands?

1. Move into your `/home/emmanuel324/projects/moose` directory and run `git log -n 1`

2. After you run `./scripts/update_and_rebuild_wasp.sh`, move into `framework/contrib/wasp` and run `git log -n 1`

**Comment by Emmanuel-324** [comment]
I started everything all over and it passes the build and test now.
